lib/src/local_working_copy.rs

```diff
@@ -1545,7 +1545,19 @@ impl FileSnapshotter<'_> {
         materialized_conflict_data: Option<MaterializedConflictData>,
     ) -> Result<MergedTreeValue, SnapshotError> {
         if let Some(current_tree_value) = current_tree_values.as_resolved() {
-            let id = self.write_file_to_store(repo_path, disk_path).await?;
+            let id = self
+                .write_file_to_store(
+                    repo_path,
+                    current_tree_value.as_ref().and_then(|tree_value| {
+                        if let TreeValue::File { id, .. } = tree_value {
+                            Some(id)
+                        } else {
+                            None
+                        }
+                    }),
+                    disk_path,
+                )
+                .await?;
```

Comment:
Another idea is to remember whether we applied EOL translation to the checked-out content. I'm not sure if the problem can be fully addressed, but the implementation would be simpler.

```
pub struct FileState {
    ...
    pub eol: bool, // or enum?
}
```

Reply:
Changed to this implementation. I don't think the implementation is necessarily simpler, because the logic to correctly update `FileState::has_crlf_in_store` scatters in 6 different places in `local_working_copy.rs`:

* 2 different locations in `FileSnapshotter::write_path_to_store`, one for files without conflict(which calls into `FileSnapshotter::write_file_to_store`), one for conflicts.
* `TreeState::write_file`
* `TreeState::write_conflict`
* 2 different locations in `TreeState::reset`, one for files without conflict, one for conflicts.

However, the performance for snapshot should be better, because we don't have to read the contents from `Store` on snapshot at the cost of the performance of `TreeState::reset`, which should be rarely used anyway.